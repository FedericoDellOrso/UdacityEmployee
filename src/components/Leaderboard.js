import { connect } from "react-redux";
import Table from "react-bootstrap/Table";
import Container from "react-bootstrap/Container";
import Figure from "react-bootstrap/Figure";

const Leaderboard = (props) => {
  console.log(props.sortedUsers);
  return (
    <Container className="justify-content-center">
      <Table striped bordered hover className="m-1">
        <thead>
          <tr>
            <th>Users</th>
            <th>Answered</th>
            <th>Created</th>
          </tr>
        </thead>
        <tbody>
          {Object.values(props.sortedUsers).map((user) => {
            return (
              <tr key={user.id}>
                <td>
                  <Figure>
                    <Figure.Image width={40} height={40} src={user.avatarURL} />
                  </Figure>
                  {user.name}
                </td>
                <td>{Object.keys(user.answers).length}</td>
                <td>{user.questions.length}</td>
              </tr>
            );
          })}
        </tbody>
      </Table>
    </Container>
  );
};

const mapStateToProps = ({ users }) => {
  const sortedUsers = Object.values(users)
    .map((user) => {
      return {
        ...user,
        tot: user.questions.length + Object.keys(user.answers).length,
      };
    })
    .sort((a, b) => a.tot < b.tot);
  console.log("Sorted", sortedUsers);
  return { sortedUsers };
};

export default connect(mapStateToProps)(Leaderboard);
