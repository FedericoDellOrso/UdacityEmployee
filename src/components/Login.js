import { connect } from "react-redux";
import { Dropdown, DropdownButton } from "react-bootstrap";
import {setAuthedUser} from "./../actions/authedUser";
import Figure from "react-bootstrap/Figure";

const Login = (props) => {
  const handleDropClick = (event) => {
    console.log("chose:",event);
    props.setAuthedUser(event);

  };
  return (
    <div>
      <h1>Employee Polls</h1>
      <Figure>
            <Figure.Image
              width={170}
              height={180}
              src="/images/employee.png"
            />
          </Figure>
      <h1>Log In</h1>

        <DropdownButton
          variant="primary"
          id="dropdown-login"
          title="Choose User"
          onSelect={handleDropClick}
          data-testid="dropdown"
        >
          {props.users.map((user) => {
            return (
              <Dropdown.Item key={user.id} eventKey={user.id}>
                {user.name}
              </Dropdown.Item>
            );
          })}
          </DropdownButton>
    </div>
  );
};
const mapStateToProps = ({ users }) => ({ users: Object.values(users) });

export default connect(mapStateToProps,{setAuthedUser})(Login);
