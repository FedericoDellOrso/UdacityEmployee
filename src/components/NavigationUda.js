import { connect } from "react-redux";
import Container from "react-bootstrap/Container";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { Link } from "react-router-dom";
import Button from "react-bootstrap/Button";
import { setAuthedUser } from "../actions/authedUser";

const NavigationUda = (props) => {
    const handleClick = (event) => {
        event.preventDefault();
        props.setAuthedUser(null);
    }
  return (
    <Navbar bg="light" variant="light">
      <Container>
        <Navbar.Brand>Employee Polls</Navbar.Brand>

        <Nav className="me-auto">
          <Nav.Link as={Link} to="/">
            Home
          </Nav.Link>
          <Nav.Link as={Link} to="/leaderboard" data-testid="Leaderboard">
            Leaderboard
          </Nav.Link>
          <Nav.Link as={Link} to="/add">
            New
          </Nav.Link>
        </Nav>
        <Nav className="justify-content-end">
          <Navbar.Text>Signed in as: {props.authedUser}</Navbar.Text>
          <Button variant="outline-secondary" size="sm" className="m-2" onClick={handleClick}>
            LogOut
          </Button>
        </Nav>
      </Container>
    </Navbar>
  );
};

const mapStateToProps = ({ authedUser }) => ({ authedUser });

export default connect(mapStateToProps,{setAuthedUser})(NavigationUda);
