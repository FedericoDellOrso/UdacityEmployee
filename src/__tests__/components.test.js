import { fireEvent, render } from "@testing-library/react";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import React from "react";
import { Provider } from "react-redux";
import { store } from "../index";
import NavigationUda from "../components/NavigationUda";
import Login from "../components/Login";
import Leaderboard from "../components/Leaderboard";
import NewQuestion from "../components/NewQuestiion";

describe("App test", () => {

  it("Test NewPoll Form", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <NavigationUda />
          <Routes>
            <Route path="/" element={<NewQuestion />} />
          </Routes>
        </Provider>
      </MemoryRouter>
    );
    const formopt1 = component.getByTestId("formopt1");
    const formopt2 = component.getByTestId("formopt2");
    fireEvent.change(formopt1, { target: { value: "sea" } });
    expect(formopt1.value).toBe("sea");
    fireEvent.change(formopt2, { target: { value: "mountain" } });
    expect(formopt2.value).toBe("mountain");
  });

  it("Test Navbar", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <NavigationUda />
        </Provider>
      </MemoryRouter>
    );

    const leaderboard = component.getByTestId("Leaderboard");
    expect(leaderboard).toBeInTheDocument();
  });

  it("Test leaderboard", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Routes>
            <Route path="/" element={<Leaderboard />} />
          </Routes>
        </Provider>
      </MemoryRouter>
    );

    const users = component.getByTestId("users");
    expect(users).toBeInTheDocument();
  });

  it("Test leaderboard snapshot", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Routes>
            <Route path="/" element={<NewQuestion />} />
          </Routes>
        </Provider>
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });

  it("Test LOGIN", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );
    const dropdown = component.getByTestId("dropdown");
    expect(dropdown).toBeInTheDocument();
  });

  it("Test NewPoll Form", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <NavigationUda />
          <Routes>
            <Route path="/" element={<NewQuestion />} />
          </Routes>
        </Provider>
      </MemoryRouter>
    );
    const newquestiontitle = component.getByTestId("newquestiontitle");
    expect(newquestiontitle).toBeInTheDocument();
  });
});
