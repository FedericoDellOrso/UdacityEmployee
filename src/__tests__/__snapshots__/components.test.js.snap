// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test Test leaderboard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1
          data-testid="newquestiontitle"
        >
          Would You Rather
        </h1>
        <h3>
          Create Your Own Poll
        </h3>
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="firstOption"
            >
              First Option
            </label>
            <input
              class="form-control"
              data-testid="formopt1"
              id="firstOption"
              placeholder="Option One"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="secondOption"
            >
              Second Option
            </label>
            <input
              class="form-control"
              data-testid="formopt2"
              id="secondOption"
              placeholder="Option Two"
              type="text"
              value=""
            />
          </div>
          <button
            class="btn btn-primary"
            data-testid="buttonpoll"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1
        data-testid="newquestiontitle"
      >
        Would You Rather
      </h1>
      <h3>
        Create Your Own Poll
      </h3>
      <form
        class=""
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="firstOption"
          >
            First Option
          </label>
          <input
            class="form-control"
            data-testid="formopt1"
            id="firstOption"
            placeholder="Option One"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="secondOption"
          >
            Second Option
          </label>
          <input
            class="form-control"
            data-testid="formopt2"
            id="secondOption"
            placeholder="Option Two"
            type="text"
            value=""
          />
        </div>
        <button
          class="btn btn-primary"
          data-testid="buttonpoll"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
